### PR TITLE
[new release] producer (0.1.0)

### PR DESCRIPTION
**Pull request by Sixstring982**
Accumulate results using monadic dependency graphs

- Project page: <a href="https://github.com/sixstring982/producer">https://github.com/sixstring982/producer</a>
- Documentation: <a href="https://sixstring982.github.io/producer">https://sixstring982.github.io/producer</a>

##### CHANGES:

* Initial release

### CODE REVIEW

**Comment by Sixstring982** [comment]
Looks like this didn't work! Let me figure out these CI errors and open another PR.

**Comment by tjm165** [comment]
> Looks like this didn't work! Let me figure out these CI errors and open another PR.
> 
> [EDIT]: Actually looks like this is probably going to be fixed by #24579. I'll keep this open in that case.

Hmm, my pull request has a necessary change but it didn't solve the CI issues. Do we have a history of when `opam-ci` started failing?

**Comment by Sixstring982** [comment]
Not sure -- it looks like it hasn't had a particularly clean history. I'm not too sure what else needs to be done here!

**Comment by avsm** [comment]
Sorry about the CI noise; all the builds restarted now on this PR.

File: packages/producer/producer.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+synopsis: "Accumulate results using monadic dependency graphs"
+maintainer: ["sixstring982@gmail.com"]
+authors: ["Trent Small"]
+license: "BSD-3-Clause"
+tags: ["dependency injection" "graph" "node" "producer"]
+homepage: "https://github.com/sixstring982/producer"
+doc: "https://sixstring982.github.io/producer"
+bug-reports: "https://github.com/sixstring982/producer/issues"
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/producer/producer.0.1.0/opam:11]:
```suggestion
  "ocaml" {>= "4.08"}
```
due to
```

#=== ERROR while compiling producer.0.1.0 =====================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/Sixstring982/producer/releases/download/v0.1.0/producer-0.1.0.tbz)
# path                 ~/.opam/4.07/.opam-switch/build/producer.0.1.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p producer -j 71 @install
# exit-code            1
# env-file             ~/.opam/log/producer-7-384c47.env
# output-file          ~/.opam/log/producer-7-384c47.out
### output ###
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.producer.objs/byte -I /home/opam/.opam/4.07/lib/hmap -intf-suffix .ml -no-alias-deps -o lib/.producer.objs/byte/producer.cmo -c -impl lib/producer.ml)
# File "lib/producer.ml", line 11, characters 8-11:
# Error: Syntax error: operator expected.
# (cd _build/default && /home/opam/.opam/4.07/bin/ocamlopt.opt -w -40 -g -I lib/.producer.objs/byte -I lib/.producer.objs/native -I /home/opam/.opam/4.07/lib/hmap -intf-suffix .ml -no-alias-deps -o lib/.producer.objs/native/producer.cmx -c -impl lib/producer.ml)
# File "lib/producer.ml", line 11, characters 8-11:
# Error: Syntax error: operator expected.
```

**Review by haochenx — APPROVED** [review]
CI now passes, and this PR looks all good to me!

Just FYI @Sixstring982, you may want to upstream https://github.com/ocaml/opam-repository/pull/24582#discussion_r1348613963 by changing the `ocaml` dep spec of the following line to `(ocaml (>= 4.08))`
https://github.com/Sixstring982/producer/blob/be96cf105e3a1dcaaf4dfcfd12267e442f86a387/dune-project#L21

**Comment by mseri** [comment]
Thanks

**Comment by Sixstring982** [comment]
@haochenx Ah, good point -- will do. Thanks for letting me know!

**Pull request merged**